### PR TITLE
Update AzureRm.ApiManagement module to 4.0.3 nuget with bug fixes

### PR DESCRIPTION
**Pull request by solankisamir**
## Description

The PR fixes the following issues in the cmdlets due to issues in .net client
https://github.com/Azure/azure-powershell/issues/6370
https://github.com/Azure/azure-powershell/issues/6515
https://github.com/Azure/azure-rest-api-specs/issues/3260
https://github.com/Azure/azure-powershell/issues/6560

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by maddieclayton** [comment]
@solankisamir You will need update the Netcore.csproj with the same SDK bump as you made to the regular .csproj file.  Same for the Test.Netcore.csproj file.

**Comment by solankisamir** [comment]
updated the .net core project

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands.ApiManagement.ServiceManagement.csproj [hunk 1]
```diff
@@ -63,10 +63,22 @@
       <HintPath>..\..\..\packages\Microsoft.Data.OData.5.8.2\lib\net40\Microsoft.Data.OData.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (maddieclayton) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands.ApiManagement.ServiceManagement.csproj]: Remove Microsoft.Rest.ClientRuntime and Microsoft.Rest.ClientRuntime.Azure, as these are imported by common dependencies.

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/packages.config [hunk 2]
```diff
@@ -1,9 +1,12 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="AutoMapper" version="6.0.2" targetFramework="net452" />
-  <package id="Microsoft.Azure.Management.ApiManagement" version="4.0.2-preview" targetFramework="net452" />
+  <package id="Microsoft.Azure.Management.ApiManagement" version="4.0.3-preview" targetFramework="net452" />
   <package id="Microsoft.Data.Edm" version="5.8.2" targetFramework="net452" />
   <package id="Microsoft.Data.OData" version="5.8.2" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.11" targetFramework="net452" />
```
Comment (maddieclayton) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/packages.config]: Remove Microsoft.Rest.ClientRuntime and Microsoft.Rest.ClientRuntime.Azure, as these are imported by common dependencies.

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.Test/Commands.ApiManagement.Test.csproj [hunk 3]
```diff
@@ -83,6 +83,14 @@
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\..\packages\Microsoft.IdentityModel.Clients.ActiveDirectory.2.28.3\lib\net45\Microsoft.IdentityModel.Clients.ActiveDirectory.WindowsForms.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (maddieclayton) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.Test/Commands.ApiManagement.Test.csproj]: Remove Microsoft.Rest.ClientRuntime and Microsoft.Rest.ClientRuntime.Azure, as these are imported by common dependencies.

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.Test/packages.config [hunk 4]
```diff
@@ -14,6 +14,8 @@
   <package id="Microsoft.Bcl.Build" version="1.0.14" targetFramework="net45" />
   <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="2.28.3" targetFramework="net45" />
   <package id="Microsoft.Net.Http" version="2.2.28" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.11" targetFramework="net452" />
```
Comment (maddieclayton) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.Test/packages.config]: Remove Microsoft.Rest.ClientRuntime and Microsoft.Rest.ClientRuntime.Azure, as these are imported by common dependencies.

File: src/ResourceManager/ApiManagement/Commands.ApiManagement/ChangeLog.md [hunk 5]
```diff
@@ -18,7 +18,12 @@
         - Additional information about change #1
 -->
 ## Current Release
-* Updated help files to include full parameter types.
```
Comment (maddieclayton) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement/ChangeLog.md]: Please add this removed line back in

File: src/ResourceManager/ApiManagement/Commands.ApiManagement/Commands.ApiManagement.csproj [hunk 6]
```diff
@@ -65,10 +65,22 @@
     <Reference Include="Microsoft.Data.Services.Client, Version=5.8.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <HintPath>..\..\..\packages\Microsoft.Data.Services.Client.5.8.2\lib\net40\Microsoft.Data.Services.Client.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (maddieclayton) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement/Commands.ApiManagement.csproj]: Remove Microsoft.Rest.ClientRuntime and Microsoft.Rest.ClientRuntime.Azure, as these are imported by common dependencies.

File: src/ResourceManager/ApiManagement/Commands.ApiManagement/packages.config [hunk 7]
```diff
@@ -1,9 +1,12 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="AutoMapper" version="6.0.2" targetFramework="net452" />
-  <package id="Microsoft.Azure.Management.ApiManagement" version="4.0.2-preview" targetFramework="net452" />
+  <package id="Microsoft.Azure.Management.ApiManagement" version="4.0.3-preview" targetFramework="net452" />
   <package id="Microsoft.Data.Edm" version="5.8.2" targetFramework="net452" />
   <package id="Microsoft.Data.OData" version="5.8.2" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.11" targetFramework="net452" />
```
Comment (maddieclayton) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement/packages.config]: Remove ClientRuntime and ClientRuntime.Azure

File: src/ResourceManager/ApiManagement/Commands.SMAPI.Test/Commands.ApiManagement.ServiceManagement.Test.csproj [hunk 8]
```diff
@@ -83,6 +83,14 @@
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\..\packages\Microsoft.IdentityModel.Clients.ActiveDirectory.2.28.3\lib\net45\Microsoft.IdentityModel.Clients.ActiveDirectory.WindowsForms.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (maddieclayton) [inline src/ResourceManager/ApiManagement/Commands.SMAPI.Test/Commands.ApiManagement.ServiceManagement.Test.csproj]: Remove ClientRuntime and ClientRuntime.Azure

File: src/ResourceManager/ApiManagement/Commands.SMAPI.Test/packages.config [hunk 9]
```diff
@@ -15,6 +15,8 @@
   <package id="Microsoft.Bcl.Build" version="1.0.14" targetFramework="net45" />
   <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="2.28.3" targetFramework="net45" />
   <package id="Microsoft.Net.Http" version="2.2.28" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.11" targetFramework="net452" />
```
Comment (maddieclayton) [inline src/ResourceManager/ApiManagement/Commands.SMAPI.Test/packages.config]: Remove ClientRuntime and ClientRuntime.Azure

File: src/ResourceManager/ApiManagement/Commands.ApiManagement/help/Add-AzureRmApiManagementApiToProduct.md [hunk 10]
```diff
@@ -37,7 +37,7 @@ This command adds the specified API to the specified product.
 Specifies the ID of an API to add to a product.
 
 ```yaml
-Type: System.String
```
Comment (maddieclayton) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement/help/Add-AzureRmApiManagementApiToProduct.md]: We need these parameter types to be full name types.  Please update the help using this:

Get-ChildItem <path to repo>/src/ResourceManager/ApiManagement/Commands.ApiManagement/help | Update-MarkdownHelp -AlphabeticParamsOrder -UseFullTypeName

I am working with platyPS to get this parameter into Update-MarkdownHelpModule, and will update the help generation doc as soon as that is release.  Thanks!

**Comment by solankisamir** [comment]
@maddieclayton  fixed.

File: src/ResourceManager/ApiManagement/Commands.ApiManagement/help/Add-AzureRmApiManagementRegion.md [hunk 11]
```diff
@@ -117,7 +117,7 @@ Valid values are:
 Type: System.Nullable`1[Microsoft.Azure.Commands.ApiManagement.Models.PsApiManagementSku]
 Parameter Sets: (All)
 Aliases:
-Accepted values: Developer, Standard, Premium, Basic
```
Comment (maddieclayton) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement/help/Add-AzureRmApiManagementRegion.md]: Why are these accepted values being removed? I still see them here: https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/ApiManagement/Commands.ApiManagement/Models/PsApiManagementSku.cs

I see this also in all the files below (I went through the first couple but assume it's generally true) - why are they being removed?

Comment (solankisamir) [reply src/ResourceManager/ApiManagement/Commands.ApiManagement/help/Add-AzureRmApiManagementRegion.md]: Looks like some issue with platyPS module, I just ran the command again 
`Update-MarkdownHelp -Path D:\github\azure-powershell\src\ResourceManager\ApiManagement\Commands.ApiManagement\help -UseFullTypeName -AlphabeticParamsOrder`

Comment (maddieclayton) [reply src/ResourceManager/ApiManagement/Commands.ApiManagement/help/Add-AzureRmApiManagementRegion.md]: That's very odd.  Did you do a full build before running the update-markdownhelp command?  If that doesn't fix it, can you just manually revert? We don't want bad information getting there, and I can investigate the platyPS issue after you get this PR in.

Comment (solankisamir) [reply src/ResourceManager/ApiManagement/Commands.ApiManagement/help/Add-AzureRmApiManagementRegion.md]: reverted using git, rerunning Import-Module after clean build did not help. 

---
In reply to: [203885613](https://github.com/Azure/azure-powershell/pull/6697#discussion_r203885613) [](ancestors = 203885613)

**Comment by solankisamir** [comment]
reverted all doc file using the cool git feature, revert all files in a directory to be like the remote branch [here](https://stackoverflow.com/a/43545012/5906211)

**Pull request merged**